### PR TITLE
Load obsolete metadata from Mongo for search results

### LINKED ISSUE #6061: Add async metadata processing info to search results on post process
We are indexing the entities after marking their metadata as obsolete in order to have that information in the search results and be able to show in the cards.
We found that it is a big bottle neck in the process of updating templates, so we are moving away from that approach.
Tests show that just loading that information from Mongo on search post-processing is much more efficient and doesn't introduce perceptible performance degradation.

### PR DESCRIPTION
**Pull request by fnocetti**
fixes #6061

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/common.v2/database/specs/MongoResultSet.spec.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ afterAll(async () => {
   await testingEnvironment.tearDown();
 });
 
+// eslint-disable-next-line max-statements
```
Comment (LaszloKecskes) [inline app/api/common.v2/database/specs/MongoResultSet.spec.ts]: Eslint line.

**Review by LaszloKecskes — APPROVED** [review]
Looks good, with one eslint line remaining. :)

However, on the issue writing side, please consider adding a short description of what performance measurements were done and how, and how things were improved after the change. Things like dataset, actions measured, measurement methodology (even if it's just console logs in the code manually),  the results thereof (both before and after).

**Comment by fnocetti** [comment]
> Looks good, with one eslint line remaining. :)
> 
> However, on the issue writing side, please consider adding a short description of what performance measurements were done and how, and how things were improved after the change. Things like dataset, actions measured, measurement methodology (even if it's just console logs in the code manually), the results thereof (both before and after).

Like I've said before, there is a document covering most of those concerns. Some other details, like highlights regarding findings, are also included in the parent issue.

**Comment by fnocetti** [comment]
`regression_suites` failure related to #6073

**Pull request merged**